stacker/blueprints/rds/base.py

```diff
@@ -0,0 +1,336 @@
+DBINSTANCE_NO_IOPS = "RDSDBInstanceNoIops"
+DBINSTANCE_WITH_IOPS = "RDSDBInstanceWithIops"
+DNS_RECORD = "DBInstanceDnsRecord"
+
+
+def common_parameters():
+    """ Return common parameters for all RDS stacks.
+
+    Returns:
+        dict: A dictionary of parameter definitions.
+    """
+
+    parameters = {
+        "VpcId": {
+            "type": "AWS::EC2::VPC::Id",
+            "description": "Vpc Id"},
+        "Subnets": {
+            "type": "List<AWS::EC2::Subnet::Id>",
+            "description": "Subnets to deploy RDS instance in."},
+        "InstanceType": {
+            "type": "String",
+            "description": "AWS RDS Instance Type",
+            "default": "db.m3.large"},
+        "AllowMajorVersionUpgrade": {
+            "type": "String",
+            "description": "Set to 'true' to allow major version "
+                           "upgrades.",
+            "default": "false",
+            "allowed_values": ["true", "false"]
+        },
+        "AutoMinorVersionUpgrade": {
+            "type": "String",
+            "description": "Set to 'true' to allow minor version upgrades "
+                           "during maintenance windows.",
+            "default": "false",
+            "allowed_values": ["true", "false"]
+        },
+        "AllocatedStorage": {
+            "type": "Number",
+            "description": "Space, in GB, to allocate to RDS instance. If "
+                           "IOPS is set below, this must be a minimum of "
+                           "100 and must be at least 1/10th the IOPs "
+                           "setting.",
+            "default": "10"},
+        "StorageEncrypted": {
+            "type": "String",
+            "description": "Set to 'false' to disable encrypted storage.",
+            "default": "true",
+            "allowed_values": ["true", "false"]
+        },
+        "IOPS": {
+            "type": "Number",
+            "description": "If set, uses provisioned IOPS for the "
+                           "database. Note: This must be no more than "
+                           "10x of AllocatedStorage. Minimum: 1000",
+            "max_value": 10000,
+            "default": "0"},
+        "InternalZoneId": {
+            "type": "String",
+            "default": "",
+            "description": "Internal zone Id, if you have one."},
+        "InternalZoneName": {
+            "type": "String",
+            "default": "",
+            "description": "Internal zone name, if you have one."},
+        "InternalHostname": {
+            "type": "String",
+            "default": "",
+            "description": "Internal domain name, if you have one."},
+    }
+
+    return parameters
+
+
+class MasterInstance(Blueprint):
+    """ Blueprint for a generic Master RDS Database Instance.
+
+    Subclasses should be created for each RDS engine for better validation of
+    things like engine version.
+    """
+
+    ENGINE = None
+
+    def _get_engine_versions(self):
```

Comment:
if something should be overridden, i usually don't put an underscore since it implies some sort of public api